File: reactapp4.client/src/interfaces/SportRadarMLBPBPPitchEvent.ts
export interface SportRadarMLBPBPPitchEvent {
  game_id: string | null;                 // uuid
  at_bat_id: string | null;               // uuid
  inning: number | null;
  half: string | null;

  hit_location: string | null;
  hit_type: string | null;
  status: string | null;
  event_id: string | null;                // uuid-ish
  outcome_id: string | null;
  created_at: string | null;              // ISO with TZ
  updated_at: string | null;              // ISO with TZ
  sequence_number: number | null;
  official: boolean | null;
  type: string | null;

  wall_clock_start_time: string | null;   // ISO with TZ
  wall_clock_end_time: string | null;     // ISO with TZ

  is_ab_over: boolean | null;
  is_bunt: boolean | null;
  is_hit: boolean | null;
  is_wild_pitch: boolean | null;
  is_passed_ball: boolean | null;
  is_double_play: boolean | null;
  is_triple_play: boolean | null;

  balls: number | null;
  strikes: number | null;
  outs: number | null;

  pitch_count: number | null;
  pitch_type: string | null;
  pitch_speed: number | null;
  pitch_zone: number | null;

  pitcher_hand: string | null;
  hitter_hand: string | null;
  pitcher_id: string | null;              // uuid
  pitch_x: number | null;
  pitch_y: number | null;

  pitcher_preferred_name: string | null;
  pitcher_first_name: string | null;
  pitcher_last_name: string | null;
  pitcher_jersey_number: number | null;
  pitcher_full_name: string | null;

  hitter_preferred_name: string | null;
  hitter_first_name: string | null;
  hitter_last_name: string | null;
  hitter_jersey_number: number | null;
  hitter_full_name: string | null;
  hitter_id: string | null;               // uuid

  home_team_runs: number | null;
  away_team_runs: number | null;

  mlb_pitch_speed: number | null;
  mlb_strike_zone_top: number | null;
  mlb_strike_zone_bottom: number | null;
  mlb_pitch_zone: number | null;
  mlb_pitch_code: string | null;
  mlb_pitch_description: string | null;
  mlb_pitch_x: number | null;
  mlb_pitch_y: number | null;
  mlb_hit_trajectory: string | null;
  mlb_hit_hardness: string | null;
  mlb_hit_x: number | null;
  mlb_hit_y: number | null;
}

File: ReactApp4.Server/Data/SportRadarMLBPBPPitchEvent.cs
using System;

namespace ReactApp4.Server.Data
{
    public class SportRadarMLBPBPPitchEvent
    {
        public Guid? GameId { get; set; }
        public Guid? AtBatId { get; set; }
        public int? Inning { get; set; }
        public string? Half { get; set; }                      // "T"/"B"

        public string? HitLocation { get; set; }
        public string? HitType { get; set; }
        public string? Status { get; set; }
        public Guid? EventId { get; set; }                     // looks like a UUID in sample
        public string? OutcomeId { get; set; }                 // "kKL" (not a UUID)
        public DateTimeOffset? CreatedAt { get; set; }
        public DateTimeOffset? UpdatedAt { get; set; }
        public int? SequenceNumber { get; set; }
        public bool? Official { get; set; }
        public string? Type { get; set; }

        public DateTimeOffset? WallClockStartTime { get; set; }
        public DateTimeOffset? WallClockEndTime { get; set; }

        public bool? IsAbOver { get; set; }
        public bool? IsBunt { get; set; }
        public bool? IsHit { get; set; }
        public bool? IsWildPitch { get; set; }
        public bool? IsPassedBall { get; set; }
        public bool? IsDoublePlay { get; set; }
        public bool? IsTriplePlay { get; set; }

        public int? Balls { get; set; }
        public int? Strikes { get; set; }
        public int? Outs { get; set; }

        public double? PitchCount { get; set; }                // sample shows 1.0
        public string? PitchType { get; set; }                 // "FA"
        public double? PitchSpeed { get; set; }                // 92.5
        public double? PitchZone { get; set; }                 // 11.0

        public string? PitcherHand { get; set; }               // "R"
        public string? HitterHand { get; set; }                // "L"
        public Guid? PitcherId { get; set; }
        public double? PitchX { get; set; }                    // 129.0
        public double? PitchY { get; set; }                    // 39.0

        public string? PitcherPreferredName { get; set; }
        public string? PitcherFirstName { get; set; }
        public string? PitcherLastName { get; set; }
        public int? PitcherJerseyNumber { get; set; }
        public string? PitcherFullName { get; set; }

        public string? HitterPreferredName { get; set; }
        public string? HitterFirstName { get; set; }
        public string? HitterLastName { get; set; }
        public int? HitterJerseyNumber { get; set; }
        public string? HitterFullName { get; set; }
        public Guid? HitterId { get; set; }

        public int? HomeTeamRuns { get; set; }
        public int? AwayTeamRuns { get; set; }

        public double? MlbPitchSpeed { get; set; }             // 92.5
        public double? MlbStrikeZoneTop { get; set; }          // 3.2235…
        public double? MlbStrikeZoneBottom { get; set; }       // 1.4902…
        public double? MlbPitchZone { get; set; }              // 11.0
        public string? MlbPitchCode { get; set; }              // "FF"
        public string? MlbPitchDescription { get; set; }       // "Four-Seam Fastball"
        public double? MlbPitchX { get; set; }                 // 151.69
        public double? MlbPitchY { get; set; }                 // 166.65
        public string? MlbHitTrajectory { get; set; }          // may be empty
        public string? MlbHitHardness { get; set; }            // may be empty
        public double? MlbHitX { get; set; }                   // may be empty
        public double? MlbHitY { get; set; }                   // may be empty
    }
}
